src/osyris/spatial/subdomain.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/osyris-project/osyris)
+
+import numpy as np
+
+
+def extract_sphere(dataset, radius, origin):
+    """
+    Extract a spherical subdomain around an origin point.
+    """
+    subdomain = dataset.__class__(nout=dataset.meta["nout"], path=dataset.meta["path"])
```

Comment:
I still think we shouldn't read the metadata from disk again. Looking at the implementation of `Dataset`, I think that if you set `nout` to `None`, the dataset won't even have a loader, which might be good? i.e. we are effectively preventing the subdomain from being able to call `.load()`, which would read everything again from disk and it would no longer be a subdomain?

I would have `subdomain = Dataset()`, and then `subdomain.meta = dataset.meta.copy()`